src/inference/src/model_reader.cpp

```diff
@@ -15,6 +14,20 @@
 #include "transformations/utils/utils.hpp"
 
 namespace {
+ov::element::Type to_legacy_type(const ov::element::Type& legacy_type, bool input) {
+    if (input) {
+        return legacy_type == ov::element::f16 ? ov::element::f32 : legacy_type;
+    } else {
+        if (legacy_type == ov::element::i64 || legacy_type == ov::element::u64 || legacy_type == ov::element::i32 ||
+            legacy_type == ov::element::u32) {
+            return ov::element::i32;
+        } else if (legacy_type != ov::element::f32) {
```

Comment:
E.g i8, i16 will converted to f32 is that OK?

Reply:
it's legacy behavior for IR v10 and API 2.0
the logic is not changed here